components/server/src/prometheus-metrics.ts

```diff
@@ -286,3 +288,20 @@ export const jobsDurationSeconds = new prometheusClient.Histogram({
     buckets: [10, 30, 60, 2 * 60, 3 * 60, 5 * 60, 10 * 60, 15 * 60],
     labelNames: ["name"],
 });
+
+// Total requests (with label on hit/miss)
+export const redisCacheRequestsTotal = new prometheusClient.Counter({
+    name: "gitpod_redis_cache_requests_total",
+    help: "Total number of cache requests",
+    labelNames: ["hit"],
+});
+
+export function reportRedisCacheRequest(hit: boolean) {
+    redisCacheRequestsTotal.inc({ hit: String(hit) });
+}
+
+export const redisCacheLatencyHistogram = new prometheusClient.Histogram({
+    name: "gitpod_redis_cache_latency_seconds",
+    help: "Cache latency in seconds",
+    buckets: [0.01, 0.1, 0.2, 0.5, 1, 2, 5, 10],
```

Comment:
IMO, it would make sense to label it with the resource (e.g `user`) as different objects will have different sizes, and hence different latency